### ISSUE TITLE
A question about hyper parameters used in your paper

### ISSUE THREAD

**Issue by LXXXXR** [mansicer/MAIC#6]
Thank you for your awesome work! 

I was trying to reproduce Fig. 5 in your AAAI paper, and I used the default parameters for the MAIC algorithm on map 27m_vs_30m with a tiny change (`batch_size=16, buffer_size=16` due to my GPU resource limitation, `seed=0; seed=1`), and I tried two sets of learning rates (`lr=0.0005, critic_lr=0.0005` as in the default configs and `lr=0.00025, critic_lr=0.00025` since I reduce the batch size).

But after 2M environment timesteps, I still got the winning rate of 0 for both training and testing for all the settings I mentioned above. The return median is somewhere between 3 and 5. So I was wondering if there is some discrepancy between the default parameters in this repository and the ones used in Fig. 5 in your paper. It would be great if you could kindly release those settings.

Thank you very much!

**Comment by mansicer** [comment]
Hello. I remember that we do not use different hyper-parameters for this map, so you should get a result close to the paper figure with a default config setting. To verify this, I run one experiment with the following script:

```bash
python src/main.py --config=maic --env-config=sc2 with env_args.map_name=27m_vs_30m seed=42
```

Currently I have got a test win rate more than 30% and a median test return near 17 with 1M time steps. I believe that this can work as long as your package versions (like numpy and torch) are not totally different from mine. It will take up to 20G graphical memories on your GPU, so you may need to lower the batch size or the network size if your GPU resources are limited. Personally I suggest to keep the learning rate unchanged because RMSProp is a little sensitive to it (though I do not evaluate the impact). Alternatively, you can also run the entire program on CPU, I remember that the speed can be acceptable if you do not occupy your CPU too much.

Unfortunately, I might also not be able to evaluate your settings or run more experiments due to my limited resources recently. I hope these approaches will work for you.

**Comment by LXXXXR** [comment]
Thank you for your kind reply. I just have a quick follow-up question that I want to confirm, so you used the same set of hyper-parameters for all the results in Fig. 5, right?

Yes, in fact, I tried `batch size = 32` setting, and I can already observe a very different return plot compared to the case when `batch size = 16`. So it shouldn't be an issue with package versions.
Thank you for your tips. However, as I mentioned in my original issue description, unfortunately, simply reducing the batch size and leaving all other hyper-parameters unchanged will dramatically degrade the performance. I'll try to tune other hyper-parameters and see what I got.

Thank you again for your help. And a happy new year to you :)

**Comment by mansicer** [comment]
Hi. I believe that we use the same hyper-parameters for all main experiments. I do not expect that simply changing the batch size will result in large discrepancies. It can be an interesting observation and maybe it is worth tuning (for MAIC and maybe other MARL algorithms). Please share some insights with me if you do so. Thank you and have a good day!

**Comment by LXXXXR** [comment]
Thank you very much for your reply. 

As a matter of fact, I totally agree with you. The fact that my experiments show that simply changing the batch size degrades the performance dramatically also comes as a shock to me. It is definitely something I would not expect. I'll double-check everything and try tuning. And I'll get back to you if there's anything interesting.

Have a good one ;)

**Comment by mansicer** [comment]
Hi, there! We will close this issue since it seems not to be other questions, but further comments are still welcome if other questions exist.
